File: 백준/1920.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
vector <int> A;
vector <int> B;
int binarySearch(int low, int high, int key) {
	while (low <= high) {
		int mid = (low + high) / 2;
		if (A[mid] == key) return 1;
		else if (A[mid] > key) high = mid - 1;
		else low = mid + 1;
	}
	return 0;
}
int main() {
	int N, M;
	cin >> N;
	A.resize(N);
	for (int i = 0; i < N; i++) {
		scanf_s("%d", &A[i]);
	}
	sort(A.begin(), A.end());
	cin >> M;
	B.resize(N);
	for (int i = 0; i < M; i++) {
		scanf_s("%d", &B[i]);
		cout << binarySearch(0, N-1, B[i]) << "\n";
	}
}

File: 백준/1920.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
//const input = require('fs').readFileSync('./test.txt').toString().trim().split('\n');
const [N, A, M, B] = input.map(v => v.split(" ").map(x => Number(x)));
A.sort((a,b)=>a-b);
const binarySearch=(arr, left, right, target)=>{
    while (left<=right) {
        const mid=Math.floor((left+right)/2);
        if (arr[mid]===target) return 1;
        if (arr[mid]>target) right=mid-1;
        else left=mid+1;
    }
    return 0;
}
const result=B.map(b=>binarySearch(A, 0, A.length-1, b));
console.log(result.join("\n"));
